src/clib/pio_internal.h

```diff
@@ -12,6 +12,13 @@
 
 #include <config.h>
 #include <pio.h>
+#include <nc.h>
+#include <ncdispatch.h>
```

Comment:
Does this restrict the versions of netcdf we can use or prohibit building pio with just an installed netcdf (are these include files normally installed with the netcdf distributon?)

Reply:
If we release 2.3.1 now, then we will call the next release 2.4.

The 2.4 release will contain whatever header files it needs. Full compatibility for all current PIOc_ applications will be maintained. It will continue to work with all versions of netCDF 4.4.1 and after. I may have to turn off some tests so they don't activate some of the fill value bugs in those releases.

However users are strongly urged to upgrade to netcdf 4.6.0 quickly as it fixes some important fill value bugs. Specifically, 0 is mistakenly used for default fill value.

Reply:
To address your other concern, users will not have to re-build netCDF. They can use any recent version.

But they should upgrade because of the fill value bugs. Those actually affect PIO only peripherally, as PIO turns off fill mode by default. But I have some tests for PIOc_set_fill() that will probably have to be guarded in some way against old netCDF builds.

Reply:
The concern is not whether users will need to rebuild netcdf, but rather whether users will be able to build pio from an installed netcdf library - it looks like for example ncdispatch.h is not installed and so users would need a netcdf dev package.

Reply:
Users will be able to build PIO with any netCDF after 4.4.1. Whatever header files are needed will be included with PIO.